builtin/logical/totp/path_keys.go

```diff
@@ -0,0 +1,400 @@
+				Description: `The name of the key's issuing organization. Required if generate is true and url is not passed.`,
+			},
+
+			"account_name": {
+				Type:        framework.TypeString,
+				Description: `The name of the account associated with the key. Required if generate is true and url is not passed.`,
+			},
+
+			"period": {
+				Type:        framework.TypeDurationSecond,
+				Default:     30,
+				Description: `The length of time used to generate a counter for the TOTP token calculation.`,
+			},
+
+			"algorithm": {
+				Type:        framework.TypeString,
+				Default:     "SHA1",
+				Description: `The hashing algorithm used to generate the TOTP token. Options include SHA1, SHA256 and SHA512.`,
+			},
+
+			"digits": {
+				Type:        framework.TypeInt,
+				Default:     6,
+				Description: `The number of digits in the generated TOTP token. This value can either be 6 or 8.`,
+			},
+
+			"skew": {
+				Type:        framework.TypeInt,
+				Default:     0,
+				Description: `The number of delay periods that are allowed when validating a TOTP token. This value can either be 0 or 1. Only used if generate is true.`,
+			},
+
+			"qr_size": {
+				Type:        framework.TypeInt,
+				Default:     200,
+				Description: `The pixel size of the generated square QR code. Only used if generate is true.`,
+			},
+
+			"url": {
+				Type:        framework.TypeString,
+				Description: `A TOTP url string containing all of the parameters for key setup. Only used if generate is false.`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathKeyRead,
+			logical.UpdateOperation: b.pathKeyCreate,
+			logical.DeleteOperation: b.pathKeyDelete,
+		},
+
+		HelpSynopsis:    pathKeyHelpSyn,
+		HelpDescription: pathKeyHelpDesc,
+	}
+}
+
+func (b *backend) Key(s logical.Storage, n string) (*keyEntry, error) {
+	entry, err := s.Get("key/" + n)
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	var result keyEntry
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+func (b *backend) pathKeyDelete(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	err := req.Storage.Delete("key/" + data.Get("name").(string))
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (b *backend) pathKeyRead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	key, err := b.Key(req.Storage, data.Get("name").(string))
+	if err != nil {
+		return nil, err
+	}
+	if key == nil {
+		return nil, nil
+	}
+
+	// Translate algorithm back to string
+	algorithm := key.Algorithm.String()
+
+	// Return values of key
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"issuer":       key.Issuer,
+			"account_name": key.AccountName,
+			"period":       key.Period,
+			"algorithm":    algorithm,
+			"digits":       key.Digits,
```

Comment:
Somehow I looked at that block five times and missed issuer each time. Sorry! Key size was just because why not but no, I don't think it's useful, so forget it.